src/NetworkLayout.jl

```diff
@@ -1,11 +1,96 @@
 module NetworkLayout
 
+using GeometryBasics
+using Requires
+using LinearAlgebra: norm
+using Random
+
+export LayoutIterator, layout
+
+"""
+    AbstractLayout{Dim,Ptype}
+
+Abstract supertype for all layouts. Each layout `Algorithm <: AbstractLayout` needs to
+implement
+
+    layout(algo::Algorithm, adj_matrix)::Vector{Point{Dim,Ptype}}
+
+which takes the adjacency matrix representation of a network and returns a list of
+node positions. Each `Algorithm` object holds all of the necessary parameters.
+
+By implementing `layout` the algorithm also inherits the function-like property
+
+    Algorithm(; kwargs...)(adj_matrix) -> node_positions
+"""
```

Comment:
```suggestion
Abstract supertype for all layouts. Each layout `Layout <: AbstractLayout` needs to
implement

    layout(::Layout, adj_matrix)::Vector{Point{Dim,Ptype}}

which takes the adjacency matrix representation of a network and returns a list of
node positions. Each `Layout` object holds all of the necessary parameters.

By implementing `layout` the Layout also inherits the function-like property

    Layout(; kwargs...)(adj_matrix) -> node_positions
"""
```
`Algorithm` is a very broad term, and might already be defined by other packages.